pkg/networkservice/core/eventchannel/monitor_server_test.go

```diff
@@ -82,8 +85,14 @@ func TestMonitorConnectionServer_MonitorConnections(t *testing.T) {
 			assert.Nil(t, err)
 		}()
 	}
-	// Give the go functions calling server.MonitorConnections(selector,sender) a chance to run
-	<-time.After(time.Millisecond)
+	// Give all the go functions calling server.MonitorConnections(selector,sender) a chance to run
+	for {
+		c := <-eventMonitorStartCh
```

Comment:
Let's add a timeout for this loop to avoid exiting on test timeout (10m by default) in case if something is refactored :) 
```suggestion
	select {
		case c := <-eventMonitorStartCh:
		...
		case <- timeoutCh:
		...
	}
		
```